### DISCUSSION TITLE
Reimplement `AuthenticationService.authenticate()`

### DISCUSSION THREAD

**Discussion by gdpdils** [directus/directus#9641] [Archive]
The fact that `AuthenticationService.authenticate()` is missing the last version of Directus is preventing me from updating to the latest version. I have a custom authentication system that integrates with out Single Sign On and I need to retrieve / refresh `accessToken` / `refreshToken` / `expires` for a user given only its email.

An alternative / reimplementation to this is much needed:

```javascript
      const { accessToken, refreshToken, expires } =
        await authenticationService.authenticate({
          email: user.email,
        });
```

Thanks

**Comment by benhaynes** [comment]
CC @aidenfoxx @rijkvanzanten

**Comment by joselcvarela** [comment]
Hello @gustavodipietrogva 
Our authentication was refactored so now instead of `authenticate` it should be `login`. Also you will need to pass extra arguments
Check here the changes made: 
https://github.com/directus/directus/commit/084c6117b7ed3c5dc3972755426b3a3c02bb1831#diff-03b58901729f938551b47a9357d79ae1d0c62868265a05e8f9980d30b22c0e95R42
